pkg/document/json/text.go

```diff
@@ -62,10 +64,15 @@ func (t *TextValue) AnnotatedString() string {
 
 // Split splits this value by the given offset.
 func (t *TextValue) Split(offset int) RGATreeSplitValue {
-	value := t.value
-	r := []rune(value)
-	t.value = string(r[0:offset])
-	return NewTextValue(string(r[offset:]))
+	grapheme := uniseg.NewGraphemes(t.value)
+
+	var split []string
+	for idx := 0; grapheme.Next(); idx++ {
```

Comment:
```suggestion
	for grapheme.Next() {
```
@hackerwins same here..!